lhs.js

```diff
@@ -15,6 +15,24 @@ function dom(selector) {
     return new DomLhs(selector);
 }
 
+/*
+ * Of all the dom nodes selected by type() or dom(), return only
+ * the fnodes that satisfy all the predicates imposed by calls to
+ * when()
```

Comment:
I was going to suggest that, rather than passing in `predicates`, you just make this a method of Lhs and have it access `this.predicates` itself. Reading the code, I see you've even already documented it that way. :-)